### ISSUE TITLE
The realtime effect settings of a duplicated track are not independent

### ISSUE THREAD

**Issue by DavidBailes** [audacity/audacity#4169]
### Steps to reproduce

1. Create an audio track.
2. Add a realtime effect
3. Duplicate the audio in the track.
4. Open the dialog for the realtime effect for the original track, and change the value of a parameter.
5. Open the dialog for the realtime effect for the duplicated track, and observe the value of the parameter changed in the previous step

### Expected behavior

The values for the parameters for the two tracks should be different.

### Actual behavior

The values for the parameters for the two tracks are the same.

### Audacity Version

3.7

### Operating system

Windows, MacOS, Linux

**Comment by Paul-Licameli** [comment]
> Testing against PR #4325 on Windows and macOS and the issue is fixed now.

Yes but there is a good objection from @crsib in the code review about other consequences of the change for undo and redo behavior.

**Comment by chinakov** [comment]
Tested: 
- Custom VST and VST3 effects are independent
- Built-in effects are independent
